core/src/main/kotlin/behavior/channel/threads/ThreadChannelBehavior.kt

```diff
@@ -0,0 +1,77 @@
+package dev.kord.core.behavior.channel.threads
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.behavior.channel.MessageChannelBehavior
+import dev.kord.core.cache.data.toData
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.thread.ThreadChannel
+import dev.kord.core.entity.channel.thread.ThreadUser
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.rest.builder.channel.thread.ThreadModifyBuilder
+import kotlinx.coroutines.flow.Flow
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+interface ThreadChannelBehavior : MessageChannelBehavior {
+
+    val members: Flow<ThreadUser>
+        get() = supplier.getThreadMembers(id)
+
+
+    suspend fun removeUser(userId: Snowflake) {
+        kord.rest.channel.removeUserFromThread(id, userId)
+    }
+
+    suspend fun addUser(userId: Snowflake) {
+        kord.rest.channel.addUserToThread(id, userId)
+    }
+
+    suspend fun join() {
+        kord.rest.channel.joinThread(id)
+    }
+
+    suspend fun leave() {
+        kord.rest.channel.leaveThread(id)
+    }
```

Comment:
Lacks documentation. Specifically the permission requirements of some functions can be of great use here.